java/vector/src/test/java/org/apache/arrow/vector/ipc/TestArrowReaderWriter.java

```diff
@@ -135,4 +151,128 @@ public void test() throws IOException {
+    vector.set(1, "bar".getBytes(StandardCharsets.UTF_8));
+    vector.set(3, "baz".getBytes(StandardCharsets.UTF_8));
+    vector.set(4, "bar".getBytes(StandardCharsets.UTF_8));
+    vector.set(5, "baz".getBytes(StandardCharsets.UTF_8));
+    vector.setValueCount(6);
+
+    FieldVector encodedVector1A = (FieldVector) DictionaryEncoder.encode(vector, dictionary);
+    vector.close();
+
+    List<Field> fields = Arrays.asList(encodedVector1A.getField());
+    List<FieldVector> vectors = Collections2.asImmutableList(encodedVector1A);
+    VectorSchemaRoot root =  new VectorSchemaRoot(fields, vectors, encodedVector1A.getValueCount());
+
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    ArrowFileWriter writer = new ArrowFileWriter(root, provider, newChannel(out));
+    writer.start();
+    writer.end();
+
+    try (SeekableReadChannel channel = new SeekableReadChannel(
+        new ByteArrayReadableSeekableByteChannel(out.toByteArray()));
+        ArrowFileReader reader = new ArrowFileReader(channel, allocator)) {
+      Schema readSchema = reader.getVectorSchemaRoot().getSchema();
+      assertEquals(root.getSchema(), readSchema);
+      assertEquals(1, reader.getDictionaryVectors().size());
+      assertEquals(0, reader.getDictionaryBlocks().size());
+      assertEquals(0, reader.getRecordBlocks().size());
+    }
+  }
+
+  @Test
+  public void testEmptyStreamInStreamingIPC() throws IOException {
+
+    DictionaryProvider.MapDictionaryProvider provider = new DictionaryProvider.MapDictionaryProvider();
+    provider.put(dictionary);
+
+    VarCharVector vector = newVarCharVector("varchar", allocator);
+    vector.allocateNewSafe();
+    vector.set(0, "foo".getBytes(StandardCharsets.UTF_8));
+    vector.set(1, "bar".getBytes(StandardCharsets.UTF_8));
+    vector.set(3, "baz".getBytes(StandardCharsets.UTF_8));
+    vector.set(4, "bar".getBytes(StandardCharsets.UTF_8));
+    vector.set(5, "baz".getBytes(StandardCharsets.UTF_8));
+    vector.setValueCount(6);
+
+    FieldVector encodedVector = (FieldVector) DictionaryEncoder.encode(vector, dictionary);
+    vector.close();
+
+    List<Field> fields = Arrays.asList(encodedVector.getField());
+    List<FieldVector> vectors = Collections2.asImmutableList(encodedVector);
+    VectorSchemaRoot root =  new VectorSchemaRoot(fields, vectors, encodedVector.getValueCount());
+
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    ArrowStreamWriter writer = new ArrowStreamWriter(root, provider, newChannel(out));
+    writer.start();
+    writer.end();
+
+    try (ArrowStreamReader reader = new ArrowStreamReader(
+        new ByteArrayReadableSeekableByteChannel(out.toByteArray()), allocator)) {
+      Schema readSchema = reader.getVectorSchemaRoot().getSchema();
+      assertEquals(root.getSchema(), readSchema);
+      assertEquals(1, reader.getDictionaryVectors().size());
+      assertFalse(reader.loadNextBatch());
+    }
+  }
 }
```

Comment:
is it possible to add a test for the new case you are handling?

Reply:
Thanks, I added a test to test reading interleaved data.